state/state_test.go

```diff
@@ -4185,7 +4179,7 @@ func (s *StateSuite) TestSetModelAgentVersionFailsIfUpgrading(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 
 	err = s.State.SetModelAgentVersion(nextVersion, nil, false)
-	c.Assert(err, jc.Satisfies, stateerrors.IsUpgradeInProgressError)
+	c.Assert(errors.Is(err, stateerrors.ErrUpgradeInProgress), jc.IsTrue)
```

Comment:
it would be nice to have a checker for this
`c.Assert(err, gc.IsError, stateerrors.ErrUpgradeInProgress)`